src/Api/Controllers/CollectionsController.cs

```diff
@@ -170,9 +190,86 @@ private async Task<Collection> GetCollectionAsync(Guid id, Guid orgId)
             return collection;
         }
 
-        private async Task<bool> ManageAnyCollections(Guid orgId)
+
+        public async Task<bool> CanCreateCollection(Guid orgId, Guid collectionId)
+        {
+            if (collectionId != default)
+            {
+                return false;
+            }
+
+            return await _currentContext.CreateNewCollections(orgId);
+        }
+
+        private async Task<bool> CanEditCollectionAsync(string orgId, string collectionId) =>
+            await CanEditCollectionAsync(new Guid(orgId), new Guid(collectionId));
+        private async Task<bool> CanEditCollectionAsync(Guid orgId, Guid collectionId)
+        {
+            if (collectionId == default)
+            {
+                return false;
+            }
+
+            if (await _currentContext.EditAnyCollection(orgId))
+            {
+                return true;
+            }
+
+            if (await _currentContext.EditAssignedCollections(orgId))
+            {
+                return null != _collectionRepository.GetByIdAsync(collectionId, _currentContext.UserId.Value);
+            }
+
+            return false;
+        }
+
+        private async Task<bool> CanDeleteCollectionAsync(string orgId, string collectionId) =>
+            await CanDeleteCollectionAsync(new Guid(orgId), new Guid(collectionId));
+        private async Task<bool> CanDeleteCollectionAsync(Guid orgId, Guid collectionId)
+        {
+            if (collectionId == default)
+            {
+                return false;
+            }
+
+            if (await _currentContext.DeleteAnyCollection(orgId))
+            {
+                return true;
+            }
+
+            if (await _currentContext.DeleteAssignedCollections(orgId))
+            {
+                return null != _collectionRepository.GetByIdAsync(collectionId, _currentContext.UserId.Value);
+            }
+
+            return false;
+        }
+
+        private async Task<bool> CanViewCollectionAsync(string orgId, string collectionId) =>
+            await CanViewCollectionAsync(new Guid(orgId), new Guid(collectionId));
+        private async Task<bool> CanViewCollectionAsync(Guid orgId, Guid collectionId)
+        {
+            if (collectionId == default)
```

Comment:
Do we need this clause? Wondering if collectionId is ever default, and if so is there a harm in checking for 00000.

Reply:
It is. The way our replace methods work is by keying off if Id is default. If it is, the record is a create, if not, it's an update.

This method first checks that we're going to create the record, then validates you're allowed to (same with the `canEditCollectionAsync` method